pkg/apis/kops/cluster.go

```diff
@@ -205,6 +205,27 @@ type ClusterSpec struct {
 	RollingUpdate *RollingUpdate `json:"rollingUpdate,omitempty"`
 	// ClusterAutoscaler defines the cluster autoscaler configuration.
 	ClusterAutoscaler *ClusterAutoscalerConfig `json:"clusterAutoscaler,omitempty"`
+	// IAMRolesForServiceAccounts defines the IAM Roles for Service Accounts configuration.
+	IAMRolesForServiceAccounts *IAMRolesForServiceAccountsConfig `json:"iamRolesForServiceAccounts,omitempty"`
+}
+
+// OIDCProviderLocation determines where OIDC metadata will be stored
+type OIDCProviderLocation string
+
+const (
+	// OIDCProviderLocationAPIServer will use the k8s API server as OIDC metadata provider.
+	OIDCProviderLocationAPIServer OIDCProviderLocation = "APIServer"
+	// OIDProviderCLocationPublicDataStore will store the OIDC metadata on a publically accessible VFS.
+	OIDCLocationPublicDataStore OIDCProviderLocation = "PublicDataStore"
+	// OIDCProviderLocationNone will not store the OIDC metadata and disable any cloud OIDC provider
+	OIDCProviderLocationNone OIDCProviderLocation = "None"
+)
+
+// InstanceRoleForServiceAccountConfig defines the IRSA configuration.
+type IAMRolesForServiceAccountsConfig struct {
+	// JWKSLocation determines the location of the OIDC metadata.
+	// Valid locations are "APIServer" and "PublicDataStore".
+	OIDCProviderLocation OIDCProviderLocation `json:"oidcLocation,omitempty"`
```

Comment:
The OIDC Provider location is not specific to IRSA. The OIDC provider can be consumed by other clients. It should not be in a substruct about IAM.

Reply:
Whether workloads should consume IAM permissions through serviceaccount or node roles does arguably belong under `IAMSpec`.

Reply:
That is fair. But then we probably need to put `enabled` back into this struct as and additionally require `OIDCProviderLocation != None`. This struct is very much about enabling the IAM OIDC provider on AWS (and someone will maybe implement support for workload identity on GCE).